wallet/wallet.c

```diff
@@ -3351,104 +3351,41 @@ wallet_payment_list(const tal_t *ctx,
-	// use a small if else to set the remain string, with this method we can have small code.
-	if (payment_hash && status) {
-		stmt =
-			db_prepare_v2(wallet->db, SQL("SELECT"
-						      "  id"
-						      ", status"
-						      ", destination"
-						      ", msatoshi"
-						      ", payment_hash"
-						      ", timestamp"
-						      ", payment_preimage"
-						      ", path_secrets"
-						      ", route_nodes"
-						      ", route_channels"
-						      ", msatoshi_sent"
-						      ", description"
-						      ", bolt11"
-						      ", failonionreply"
-						      ", total_msat"
-						      ", partid"
-						      ", local_offer_id"
-						      " FROM payments"
-						      " WHERE payment_hash = ? AND status = ?"
-						      " ORDER BY id;"));
-		db_bind_sha256(stmt, 0, payment_hash);
-		db_bind_int(stmt, 1, wallet_payment_status_in_db(*status));
-	} else if (payment_hash) {
-		stmt =
-		    db_prepare_v2(wallet->db, SQL("SELECT"
-						  "  id"
-						  ", status"
-						  ", destination"
-						  ", msatoshi"
-						  ", payment_hash"
-						  ", timestamp"
-						  ", payment_preimage"
-						  ", path_secrets"
-						  ", route_nodes"
-						  ", route_channels"
-						  ", msatoshi_sent"
-						  ", description"
-						  ", bolt11"
-						  ", failonionreply"
-						  ", total_msat"
-						  ", partid"
-						  ", local_offer_id"
-						  " FROM payments"
-						  " WHERE payment_hash = ?"
-						  " ORDER BY id;"));
-		db_bind_sha256(stmt, 0, payment_hash);
-	} else if (status) {
-        	stmt = db_prepare_v2(wallet->db, SQL("SELECT"
-						     "  id"
-						     ", status"
-						     ", destination"
-						     ", msatoshi"
-						     ", payment_hash"
-						     ", timestamp"
-						     ", payment_preimage"
-						     ", path_secrets"
-						     ", route_nodes"
-						     ", route_channels"
-						     ", msatoshi_sent"
-						     ", description"
-						     ", bolt11"
-						     ", failonionreply"
-						     ", total_msat"
-						     ", partid"
-						     ", local_offer_id"
-						     " FROM payments"
-						     " WHERE status = ?"
-						     " ORDER BY id;"));
-		db_bind_int(stmt, 0, wallet_payment_status_in_db(*status));
-	} else {
-		stmt = db_prepare_v2(wallet->db, SQL("SELECT"
-						     "  id"
-						     ", status"
-						     ", destination"
-						     ", msatoshi"
-						     ", payment_hash"
-						     ", timestamp"
-						     ", payment_preimage"
-						     ", path_secrets"
-						     ", route_nodes"
-						     ", route_channels"
-						     ", msatoshi_sent"
-						     ", description"
-						     ", bolt11"
-						     ", failonionreply"
-						     ", total_msat"
-						     ", partid"
-						     ", local_offer_id"
-						     " FROM payments"
-						     " ORDER BY id;"));
-	}
-	db_query_prepared(stmt);
+
+	uint8_t enable_payment_hash = payment_hash != NULL ? 0 : 1;
+	uint8_t enable_status = status != NULL ? 0 : 1;
```

Comment:
I think we have a `u8` alias for this type, just for consistency sake, but ideally we'd have a `bool` type, which I assume you were not using here due to it missing in `db_bind_*`, is that correct?

Reply:
Oh, this is a great this, sorry if I use it, I think I bring inside some C++ convention here, sorry. But in reality, I will never understand if there is some difference from uint8_t u8 :)

I didn't use the boolean here, because I'm using SQLite in one of my side projects with android and flutter, and I know that sqlfile has no support for boolean type.

I just typed on google and I receive the following answer https://www.sqlite.org/datatype3.html

However, I don't know if I'm missing some things, I think have a boolean type is very cool, and I can change it if is supported.